src/ServantElm.hs

```diff
@@ -126,12 +165,44 @@ mkUrl segments =
           error
             "to implement - for captures, not needed now"
 
+elmTypeRefToDoc :: TypeRef -> Doc ann
+elmTypeRefToDoc = \case
+  RefPrim elmPrim -> elmPrimToDoc elmPrim
+  RefCustom (TypeName typeName) -> pretty typeName
+
+elmTypeParenDoc :: TypeRef -> Doc ann
+elmTypeParenDoc = parens . elmTypeRefToDoc
+
+elmPrimToDoc :: ElmPrim -> Doc ann
```

Comment:
This is taken from elm-street, right? You should add a comment, so that we know what we should ask elm-street to expose from their api eventually.

Reply:
done